### PR TITLE
HHH-12226 : ObjectNotFoundException thrown when @NotFound(action = NotFoundAction.IGNORE) used with enhancement

### PR DESCRIPTION
**Pull request by gbadner**
https://hibernate.atlassian.net/browse/HHH-12226

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/type/EntityType.java [hunk 1]
```diff
@@ -681,7 +681,7 @@ protected final Object resolveIdentifier(Serializable id, SharedSessionContractI
 				getAssociatedEntityName(),
 				id,
 				eager,
-				isNullable() && !isProxyUnwrapEnabled
+				isNullable()
 		);
```
Comment (gbadner) [inline hibernate-core/src/main/java/org/hibernate/type/EntityType.java]: Before the change above, ```LoadType``` would end up being ```INTERNAL_LOAD_LAZY``` when enhancement is used with ```@LazyToOne(value = LazyToOneOption.NO_PROXY)```.  See [```SessionImpl#internalLoad```](https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java#L1109-L1116).

```INTERNAL_LOAD_LAZY``` does not allow ```null``. When Hibernate tries to load the entity, it is not found and ```ObjectNotFoundException``` is thrown.

After the change, ```LoadType``` will be ```INTERNAL_LOAD_NULLABLE```, which properly returns ```null```.

Comment (gbadner) [inline hibernate-core/src/main/java/org/hibernate/type/EntityType.java]: @sebersole, is proxy unwrapping still supported w/o bytecode enhancement?

Comment (sebersole) [reply hibernate-core/src/main/java/org/hibernate/type/EntityType.java]: Sounds reasonable

Comment (sebersole) [reply hibernate-core/src/main/java/org/hibernate/type/EntityType.java]: proxy unwrapping without bytecode enhancement would just be eager loading.  so in that sense yes and no

**Comment by gbadner** [comment]
Merged.

**Pull request closed without merging**